src/core/drive/page_renderer.js

```diff
@@ -114,7 +114,12 @@ export class PageRenderer extends Renderer {
 
   removeUnusedHeadStylesheetElements() {
     for (const element of this.unusedHeadStylesheetElements) {
-      document.head.removeChild(element)
+      // Trix dynamically adds styles to the head that we want to keep around
+      // Long term we should moves those styles to Trix's CSS file
+      // but for now we'll just skip removing them
+      if (!element.dataset.tagName) {
+        document.head.removeChild(element)
```

Comment:
Just an idea, but given we're ignoring certain items based on a data attribute, I wonder if we should make a more official way to opt in to that, in case people want to have the same handling for other legacy code. We could make it so that anything with `data-turbo-permanent` doesn't get removed, and then the special case for Trix is that we also treat those with the `dataset.tagName` as if they had been marked `data-turbo-permanent`. Essentially just check 2 attributes instead of 1.

The code currently has to also special-case the styles for the Turbo progress bar. It does that by ID right now, but we could change that to have `data-turbo-permanent` too.

We could look at that in a separate PR of course, after landing this fix as-is.

Reply:
One other thing -- do we know what the specific values for that `tagName` attribute will be? Could we make the check stricter by using it?

Reply:
Since Turbo manages the progress bar, I really like the idea of rendering it with `[data-turbo-permanent]`.

Reply:
👍 makes sense, I've updated the branch so now we'll keep any styles marked with `data-turbo-permanent` and also use `data-turbo-permanent` to keep the progress bar styles around.